network/network.go

```diff
@@ -1075,3 +1131,17 @@ func (n *network) disconnected(p *peer) {
 		n.router.Disconnected(p.id)
 	}
 }
+
+// assumes stateLock is held
+// iterate the peers and find if we are already peered to this node.
+func (n *network) isPeered(id ids.ShortID) bool {
+	for _, peer := range n.peers {
```

Comment:
We should probably maintain a map instead of looping over all the peers each time